frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,154 @@
+import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/do';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { AppOffer, ApiOffer, BaseOffer } from 'app/homepage-offer/offers.model';
+import { AuthService } from 'app/auth.service';
+import { User } from 'app/user';
+import { Observable } from 'rxjs/Observable';
+import { OffersService } from 'app/homepage-offer/offers.service';
+import { ActivatedRoute, Params } from '@angular/router';
+import { environment } from 'environments/environment';
+import { Subscription } from 'rxjs/Subscription';
+import { FileReaderEvent, FileReaderEventTarget } from '../../models';
+import { NgForm, FormGroup } from '@angular/forms';
+import { FormBuilder } from '@angular/forms';
+import { Validators } from '@angular/forms';
+import { Image } from 'app/homepage-offer/image.model';
+
+@Component({
+  selector: 'volontulo-create-offer',
+  templateUrl: './create-offer.component.html',
+  styleUrls: ['./create-offer.component.scss']
+})
+export class CreateOfferComponent implements OnInit, OnDestroy {
+  public djangoRoot = environment.djangoRoot;
+  public file: File;
```

Comment:
If `file` is not used in component template, then it could be private. However, if it is not used in a template we can delete this line entirely and use the local variable in a file reading method, as I will show you in next comment.